scripts/provisioning/s3_start

```diff
@@ -0,0 +1,108 @@
+# limitations under the License.
+#
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+
+import sys
+import argparse
+import traceback
+import errno
+import logging
+import os
+import socket
+from logging import handlers
+
+#Logger details
+s3_single_entry_logger_name = \
+    "s3-deployment-logger-" + "[" + str(socket.gethostname()) + "]"
+s3_single_entry_log_file = \
+    "/var/log/seagate/s3/s3deployment/s3_single_entry.log"
+s3_single_entry_log_directory = "/var/log/seagate/s3/s3deployment/"
+s3_single_entry_log_format = \
+    "%(asctime)s - %(name)s - %(levelname)s - %(message)s"
+
+def main():
+
+  parser = argparse.ArgumentParser("S3server single entry command")
+  parser.add_argument("--service", required=True)
+
+  args = parser.parse_args()
+
+  create_logger_directory()
+  logger = create_logger()
+
+  try:
+    if args.service == 's3server':
+      # Action
+    elif args.service == 's3authserver':
+      os.system("sh /opt/seagate/cortx/auth/startauth.sh")
+    elif args.service == 's3bgproducer':
+      # Action
+    elif args.service == 's3bgconsumer':
+      # Action
+  except Exception as e:
+    logger.error(f"{str(e)}")
+    logger.error(f"{traceback.format_exc()}")
+    return errno.EINVAL
+
+  return 0
+
+def create_logger():
+    """Create logger, file handler, console handler and formatter."""
+    # create logger with "S3 deployment logger"
+    logger = logging.getLogger(s3_single_entry_logger_name)
+    logger.setLevel(logging.DEBUG)
+    # maxBytes 5242880 will allow base file s3_single_entry.log to rotate 
+    # after every 5MB. With a backupCount of 5, files will be generated 
+    # like s3_single_entry.log, s3_single_entry.log.1 and up to s3.log.5.
+    # The file being written to is always s3_single_entry.log.
+    fhandler = logging.handlers.RotatingFileHandler(s3_single_entry_log_file, \
+                                                    mode='a',\
+                                                    maxBytes = 5242880,\
+                                                    backupCount = 5,\
+                                                    encoding=None, delay=False)
+    fhandler.setLevel(logging.DEBUG)
+
+    # create console handler with a higher log level
+    chandler = logging.StreamHandler(sys.stdout)
+    chandler.setLevel(logging.DEBUG)
+
+    formatter = logging.Formatter(s3_single_entry_log_format)
+
+    # create formatter and add it to the handlers
+    fhandler.setFormatter(formatter)
+    chandler.setFormatter(formatter)
+
+    # add the handlers to the logger
+    logger.addHandler(fhandler)
+    logger.addHandler(chandler)
+
+    return logger
+
+def create_logger_directory():
+    """Create log directory if not exsists."""
+    _logger_directory = os.path.join(s3_single_entry_log_directory)
+    if not os.path.isdir(_logger_directory):
+        try:
+            os.mkdir(_logger_directory)
```

Comment:
This will fail if  parent directories don't exist.  Use `os.makedirs()`?...

Reply:
Done.